src/test/java/org/broadinstitute/consent/http/health/ElasticSearchHealthCheckTest.java

```diff
@@ -0,0 +1,99 @@
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.MockitoAnnotations;
+import org.mockserver.client.MockServerClient;
+import org.testcontainers.containers.MockServerContainer;
+
+import javax.ws.rs.InternalServerErrorException;
+import java.util.Collections;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+
+public class ElasticSearchHealthCheckTest implements WithMockServer {
+    private ElasticSearchHealthCheck healthCheck;
+    private ElasticSearchConfiguration config;
+    private RestClient client;
+    private MockServerClient mockServerClient;
+
+    @Rule
+    public MockServerContainer container = new MockServerContainer(IMAGE);
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+
+        config = new ElasticSearchConfiguration();
+        config.setServers(Collections.singletonList("localhost"));
+        config.setPort(container.getServerPort());
+        client = ElasticSearchSupport.createRestClient(config);
+
+        mockServerClient = new MockServerClient(container.getHost(), container.getServerPort());
+    }
+
+    @After
+    public void tearDown() {
+        stop(container);
+    }
+
+    private void initHealthCheck(String status, Integer statusCode) {
+        try {
+            String stringResponse = "{ \"status\": \"" + status + "\" }";
+            mockServerClient.when(request()).respond(response()
+                    .withStatusCode(statusCode)
+                    .withBody(stringResponse));
+
+            healthCheck = new ElasticSearchHealthCheck(config);
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+    }
+
+    @Test
+    public void testCheckSuccessGreen() throws Exception {
+        initHealthCheck("green", HttpStatusCodes.STATUS_CODE_OK);
+
+        HealthCheck.Result result = healthCheck.check();
+        assertTrue(result.isHealthy());
+    }
+
+    @Test
+    public void testCheckSuccessYellow() throws Exception {
+        initHealthCheck("yellow", HttpStatusCodes.STATUS_CODE_OK);
+
+        HealthCheck.Result result = healthCheck.check();
+        assertTrue(result.isHealthy());
+    }
+
+    @Test
+    public void testCheckFailureRed() throws Exception {
+        initHealthCheck("red", HttpStatusCodes.STATUS_CODE_OK);
+
+        HealthCheck.Result result = healthCheck.check();
+        assertFalse(result.isHealthy());
+    }
+
+    @Test
+    public void testCheckServerFailure() throws Exception {
+        initHealthCheck("green", HttpStatusCodes.STATUS_CODE_SERVER_ERROR);
+        try {
+            healthCheck.check();
+            fail("Expected InternalServerErrorException was not thrown.");
+        } catch (InternalServerErrorException e) {
+            return;
+        }
+    }
```

Comment:
This test should expect a failure case - it simplifies the code quite a bit
```suggestion
    @Test(expected = InternalServerErrorException.class)
    public void testCheckServerFailure() throws Exception {
        initHealthCheck("green", HttpStatusCodes.STATUS_CODE_SERVER_ERROR);
        healthCheck.check();
    }
```

Reply:
I used the syntax I learned in SG, but thanks for showing me a better one.